src/rtc/h5_prep.py

```diff
@@ -199,107 +233,238 @@ def populate_metadata_group(h5py_obj: h5py.File,
-            [orbit_files, 'List of input orbit files used'],
-        'RTC/metadata/processingInformation/inputs/auxcalFiles':
-            [[burst_in.burst_calibration.basename_cads, burst_in.burst_noise.basename_nads],
+            ['orbit_files', orbit_files, 'List of input orbit files used'],
+        'RTC/metadata/processingInformation/inputs/auxCalFiles':
+            ['aux_cal_files', [burst_in.burst_calibration.basename_cads,
+             burst_in.burst_noise.basename_nads],
              'List of input calibration files used'],
         'RTC/metadata/processingInformation/inputs/configFiles':
-            [cfg_in.run_config_path, 'List of input config files used'],
+            ['config_files', cfg_in.run_config_path,
+             'List of input config files used'],
         'RTC/metadata/processingInformation/inputs/demSource':
-            [dem_description, 'DEM source description']
+            ['dem_source', dem_description, 'DEM source description']
     }
-    for fieldname, data in dict_field_and_data.items():
+    if is_mosaic:
+        return metadata_dict
+
+    metadata_dict['identification/boundingPolygon'] = \
+        ['bounding_polygon', get_polygon_wkt(burst_in),
+         'OGR compatible WKT representation of the product bounding polygon'
+         ' bounding polygon']
+
+    metadata_dict['identification/burstID'] = \
+        ['burst_id', str(burst_in.burst_id),
+         'Burst identification (burst ID)']
+
+    beam_id = str(burst_in.burst_id)[-3:]
+    metadata_dict['identification/beamID'] = \
+        ['beam_id', beam_id, 'Beam identification (Beam ID)']
+
+    metadata_dict['identification/zeroDopplerStartTime'] = \
+        ['zero_doppler_start_time',
+         burst_in.sensing_start.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+         'Azimuth start time of the product']
+    metadata_dict['identification/zeroDopplerEndTime'] = \
+        ['zero_doppler_end_time',
+         burst_in.sensing_stop.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+         'Azimuth stop time of the product']
+    return metadata_dict
+
+
+def all_metadata_dict_to_geotiff_metadata_dict(metadata_dict):
+    '''
+    Convert all metadata dict to GeoTIFF metadata dict
+    Parameters
+    ----------
+    metadata_dict : dict
+        Metadata dict organized as follows:
+        - Dictionary item key: HDF5 dataset key;
+        - Dictionary item value: list of 
+            [GeoTIFF metadata key,
+             metadata value,
+             metadata description]
+        The value `None` for the GeoTIFF metadata key indicates that
+        the field is not saved on the GeoTIFF file
+    Returns
+    -------
+    geotiff_metadata_dict : dict
+        Metadata dict to be saved onto the RTC-S1 product GeoTIFF file
+    '''
+    geotiff_metadata_dict = {}
+    for _, (key, value, _) in metadata_dict.items():
+        if key is None:
+            continue
+        if isinstance(value, str):
+            geotiff_metadata_dict[key.upper()] = str(value).upper()
```

Comment:
Any reason that you convert the strings to uppercases?

Reply:
I'm following the convention we used for DSWx-HLS.. not sure what would be ideal. Do you prefer lowercase?

Reply:
depends on why DSWX-HLS decided to go uppercase strings, but I think it would be okay to leave as it is.